fire/inspectutils.py

```diff
@@ -22,6 +22,27 @@
 
 import IPython
 import six
+from collections import namedtuple
+
+_FullArgTuple = namedtuple('FullArgSpec', ('args', 'varargs', 'varkw',
+  'defaults', 'kwonlyargs', 'kwonlydefaults', 'annotations'))
+
+
+class _FullArgSpec(_FullArgTuple):
```

Comment:
Thinking aloud:
Is a class that inherits from a namedtuple really the best choice here?
My initial aversion is that it feels clunky. Also, now that we have 7 args, it strikes me that variable expansion is prone to breaking if we mess up the argument order somewhere and don't notice.
My preference would be to abandon the tuple aspect. And instead I think we should just have a class, and always access the members by name. Something like:

```
class FullArgSpec(object):
  """The full argspec of a callable."""

  def __init__(self, args=None, varargs=None, varkw=None, defaults=None,
               kwonlyargs=None, kwonlydefaults=None, annotations=None):
    self.args = args or []
    self.varargs = varargs
    self.varkw = varkw
    self.defaults = defaults or ()
    self.kwonlyargs = kwonlyargs or []
    self.kwonlydefaults = kwonlydefaults or {}
    self.annotations = annotations or {}
```

How's that sound?